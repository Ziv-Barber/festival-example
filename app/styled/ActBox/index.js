// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Our types:
import type { ActBoxPropsType } from './index.types'

// import IconButton from '@material-ui/core/IconButton'
// import InfoIcon from '@material-ui/icons/Info'
// import Dialog from '@material-ui/core/Dialog'
// import DialogTitle from '@material-ui/core/DialogTitle'
// <IconButton className={classes.icon}>
// clear
// forward
// <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" {...other}>
// <DialogTitle id="simple-dialog-title">Set backup account</DialogTitle>

/**
 * Control to display a single act.
 *
 * @param {ActBoxPropsType} props The properties.
 * @reactProps {string} name - The act's name.
 * @reactProps {string} image - The act's image URL.
 * @reactProps {string} className - optional CSS Class name to use for the main element of the component.
 */
const ActBox = (props: ActBoxPropsType) => (
  <div className={props.className}>
    <div className="actBoxIn">
      <p>{props.name}</p>
    </div>
  </div>
)

export default styled(ActBox)`
  position: relative;
  width: 184px;
  height: 184px;
  border: 2px solid #fff;
  transition: 0.4s ease-in-out;

  & :hover {
    transform: scale(1.1, 1.1);
  }

  & .actBoxIn {
    position: absolute;
    margin: 0;
    height: 180px;
    width: 180px;
    overflow: hidden;
  }

  & .actBoxIn:before {
    content: '';
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    background-image: url(${(props) => props.image});
    background-size: 180px 180px;
    filter: grayscale(0) blur(0);
    transition: 0.4s ease-in-out;
  }

  & .actBoxIn:hover:before {
    filter: grayscale(100%) blur(3px);
    transition: 0.4s ease-in-out;
  }

  & .actBoxIn p {
    position: absolute;
    top: -50px;
    color: #000;
    background: #ff0;
    border: 1px solid #fff;
  }

  & :hover .actBoxIn p {
    position: absolute;
    display: block;
    margin: 0;
    top: 72px;
    left: -61px;
    width: 160%;
    height: 20px;
    font-size: 16px;
    font-weight: bold;
    z-index: 1060;
    text-align: center;
    transform: rotate(45deg);
    filter: alpha(opacity=100);
    transition: 0.4s ease-in-out;
  }
`

// In case that someone need our types:
export type { ActBoxPropsType as Props }
