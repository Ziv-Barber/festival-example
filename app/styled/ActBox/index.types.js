// @flow

export type ActBoxPropsType = {
  name: string,
  image: string,
  className?: string
}
