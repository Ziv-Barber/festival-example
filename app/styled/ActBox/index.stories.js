import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import ActBox from './index'

import armstrong from '../../content/lineup/louis.jpg'

storiesOf('basic.basic.ActBox', module).add(
  'ActBox',
  () => <ActBox name="Louis Armstrong" image={armstrong} />,
  { notes: { markdown: 'This is an example to the ActBox component.' } }
)
