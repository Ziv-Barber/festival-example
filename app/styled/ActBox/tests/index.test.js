// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import ActBox from '../index'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {ActBox} */
describe('<ActBox />', () => {
  afterEach(cleanup)

  /** @test {ActBox} */
  it('should render', () => {
    const { container } = render(
      <ActBox name="This is a text" image="http://test.com/test.jpg" />
    )
    expect(container).toMatchSnapshot()
  })
})
