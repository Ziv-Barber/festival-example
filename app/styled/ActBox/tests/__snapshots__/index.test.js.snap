// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActBox /> should render 1`] = `
<div>
  <div
    class="ActBox-ph7t7j-0 eTWbgo"
  >
    <div
      class="actBoxIn"
    >
      <p>
        This is a text
      </p>
    </div>
  </div>
</div>
`;
